libarchive/archive_read_disk_entry_from_file.c

```diff
@@ -419,12 +419,28 @@ setup_acls(struct archive_read_disk *a,
 	if (accpath == NULL)
 		accpath = archive_entry_pathname(entry);
 
```

Comment:
If we don't have a fd and we are inside a subdirectory we need to set `a->tree_enter_working_dir()` otherwise `acl_get_file()` will always lookup a file in the top level directory, which remains current. This code is analogous to the code in `setup_xattrs()`.